static/app/views/performance/trends/changeExplorerUtils/spansList.tsx

```diff
@@ -0,0 +1,527 @@
+                  }
+                  projectID={projectID}
+                  location={location}
+                  organization={organization}
+                  transactionName={transaction.transaction}
+                  limit={6}
+                  isLoading={
+                    transactionsLoadingBefore ||
+                    transactionsLoadingAfter ||
+                    spansLoadingBefore ||
+                    spansLoadingAfter
+                  }
+                  isError={
+                    hasSpansErrorBefore ||
+                    hasSpansErrorAfter ||
+                    transactionsErrorBefore ||
+                    transactionsErrorAfter
+                  }
+                />
+              );
+            }}
+          </SuspectSpansQuery>
+        );
+      }}
+    </SuspectSpansQuery>
+  );
+}
+
+function updateLocation(
+  location: Location,
+  start: string,
+  end: string,
+  transaction: NormalizedTrendsTransaction,
+  projectID?: string
+) {
+  return {
+    ...location,
+    start,
+    end,
+    statsPeriod: undefined,
+    sort: SpanSortOthers.SUM_EXCLUSIVE_TIME,
+    project: projectID,
+    query: {
+      query: 'transaction:' + transaction.transaction,
+      statsPeriod: undefined,
+      start,
+      end,
+      project: projectID,
+    },
+  };
+}
+
+function updateEventView(
+  trendView: TrendView,
+  start: string,
+  end: string,
+  transaction: NormalizedTrendsTransaction,
+  sort: SpanSortOption,
+  projectID?: string
+) {
+  const newEventView = trendView.clone();
+  newEventView.start = start;
+  newEventView.end = end;
+  newEventView.statsPeriod = undefined;
+  newEventView.query = `event.type:transaction transaction:${transaction.transaction}`;
+  newEventView.project = projectID ? [parseInt(projectID, 10)] : [];
+  newEventView.additionalConditions = new MutableSearch('');
+  return newEventView
+    .withColumns(
+      [...Object.values(SpanSortOthers), ...Object.values(SpanSortPercentiles)].map(
+        field => ({kind: 'field', field})
+      )
+    )
+    .withSorts([{kind: 'desc', field: sort.field}]);
+}
+
+function findSpansNotIn(
+  initialSpans: AveragedSuspectSpan[] | undefined,
+  comparingSpans: AveragedSuspectSpan[] | undefined
+) {
+  return initialSpans?.filter(initialValue => {
+    const spanInComparingSet = comparingSpans?.filter(
+      comparingValue =>
+        comparingValue.op === initialValue.op &&
+        comparingValue.group === initialValue.group
+    );
+    return spanInComparingSet?.length === 0;
+  });
+}
+
+function findSpansIn(
+  initialSpans: AveragedSuspectSpan[] | undefined,
+  comparingSpans: AveragedSuspectSpan[] | undefined
+) {
+  return initialSpans?.filter(initialValue => {
+    const spanInComparingSet = comparingSpans?.filter(
+      comparingValue =>
+        comparingValue.op === initialValue.op &&
+        comparingValue.group === initialValue.group
+    );
+    return spanInComparingSet?.length !== 0;
+  });
```

Comment:
Same argument as above. Rather than comparing every element in both lists, stop as soon as a match is found.

```suggestion
  return initialSpans?.filter(initialValue => {
    const spanInComparingSet = comparingSpans?.find(
      comparingValue =>
        comparingValue.op === initialValue.op &&
        comparingValue.group === initialValue.group
    );
    return spanInComparingSet !== undefined
  });
```